cmd/telemetrygen/internal/metrics/worker.go

```diff
@@ -28,29 +29,63 @@ type worker struct {
 
 func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Exporter) {
 	limiter := rate.NewLimiter(w.limitPerSecond, 1)
-	var i int64
 
+	var i int64
 	for w.running.Load() {
-		rm := metricdata.ResourceMetrics{
-			Resource: res,
-			ScopeMetrics: []metricdata.ScopeMetrics{
-				{
-					Metrics: []metricdata.Metrics{
+		var metrics []metricdata.Metrics
+		if w.metricType == metricTypeGauge || w.metricType == metricTypeAll {
+			metrics = append(metrics, metricdata.Metrics{
+				Name: "gen.metric.gauge",
+				Data: metricdata.Gauge[int64]{
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							StartTime: time.Now(),
+							Time:      time.Now().Add(1 * time.Second),
+							Value:     i,
+						},
+					},
+				},
+			})
+		} else if w.metricType == metricTypeSum || w.metricType == metricTypeAll {
+			metrics = append(metrics, metricdata.Metrics{
+				Name: "gen.metric.sum",
+				Data: metricdata.Sum[int64]{
+					IsMonotonic: true,
+					Temporality: metricdata.DeltaTemporality,
+					DataPoints: []metricdata.DataPoint[int64]{
 						{
-							Name: "gen",
-							Data: metricdata.Gauge[int64]{
-								DataPoints: []metricdata.DataPoint[int64]{
-									{
-										Time:  time.Now(),
-										Value: i,
-									},
-								},
-							},
+							StartTime: time.Now(),
+							Time:      time.Now().Add(1 * time.Second),
```

Comment:
This puts the point in the future, let's do it the other way around and have the `StartTime` be one second in the past
```suggestion
							StartTime: time.Now().Add(-1 * time.Second),
							Time:      time.Now(),
```